lib/srv/desktop/rdp/rdpclient/src/client.rs

```diff
@@ -318,7 +308,18 @@ impl Client {
                             ClientResult::from(code)?;
                         }
                         ClientFunction::UpdateClipboard(data) => {
-                            *(clipboard_data.lock().unwrap()) = Some(data.clone());
+                            {
+                                let mut x224_processor = Self::x224_lock(&x224_processor)?;
+                                let cliprdr = x224_processor
+                                    .get_svc_processor_mut::<Cliprdr>()
+                                    .ok_or(ClientError::InternalError)?
+                                    .backend
+                                    .as_any_mut()
+                                    .downcast_mut::<TeleportCliprdrBackend>()
+                                    .ok_or(ClientError::InternalError)?;
+                                cliprdr.set_clipboard_data(Some(data.clone()));
+                            } // unlock x224_processor
```

Comment:
I'm unsure what this comment corresponds to. The block before it? it looks like the block before locks the `clipboard_data` (shooting in the dark here, have 0% rust knowledge)